### DISCUSSION TITLE
Pydantic v2: how to post-process the generated schema now?

### DISCUSSION THREAD

**Discussion by cherrywins** [pydantic/pydantic#8068] [Question] [answered]
On this [documentation page](https://docs.pydantic.dev/1.10/usage/schema/#schema-customization) we see the example of how we can alternatively set schema_extra to a callable and post-process the generated schema. How to do it now when class Config is deprecated?
![image](https://github.com/pydantic/pydantic/assets/25788906/b1b565d2-68c7-4301-83ac-9c8ffab5afb3)

**Comment by Viicos** [comment]
`Config` class was just replaced by `model_config`. `schema_extra` is still available: https://docs.pydantic.dev/dev/api/config/#pydantic.config.ConfigDict.json_schema_extra

**Reply by cherrywins** [reply]
It is accessible as attribute, but what if I want to post_process it? Is there any example of how to do it in Pydantic V2?

**Reply by Viicos** [reply, marked as answer]
```python
from pydantic import BaseModel, ConfigDict

def my_schema_extra(schema: dict[str, Any]) -> None:
    for prop in schema.get('properties', {}).values():
        prop.pop('title', None)


class Person(BaseModel):
    name: str
    age: int

    model_config = ConfigDict(
        json_schema_extra=my_schema_extra,
    )
```

**Reply by cherrywins** [reply]
Thank you for the help!

**Comment by ldacey** [comment]
This works with pydantic 2.3.0:

```python
def serialize_complex_types(schema: dict[str, Any]):

    for prop in schema.get("properties", {}).values():

        if "polars_type" in prop:
            prop["polars_type"] = str(prop["polars_type"])

        if "pyarrow_type" in prop:
            prop["pyarrow_type"] = str(prop["pyarrow_type"])

    return schema


class ExampleModel(SchemaModel):
    """Example schema for testing"""

    model_config = ConfigDict(json_schema_extra=serialize_complex_types)
```

But if I update to 2.5.2 it does not seem to work? I added a print statement to print the "prop" in a loop and there is no output at all in 2.5.2
